src/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -259,53 +259,56 @@ private void SetUserDetails(K8SConfiguration k8SConfig, Context activeContext)
+        /// <param name="kubeconfigPath">Explicit file path to kubeconfig. Set to null to use the default file path</param>
+        /// <returns></returns>
+        public static async Task<K8SConfiguration> LoadKubeConfigAsync(string kubeconfigPath = null)
+        {
+            var fileInfo = new FileInfo(kubeconfigPath ?? KubeConfigDefaultLocation);
+
+            return await LoadKubeConfigAsync(fileInfo);
+        }
+
+        /// <summary>
+        ///     Loads entire Kube Config from default or explicit file path
+        /// </summary>
+        /// <param name="kubeconfigPath">Explicit file path to kubeconfig. Set to null to use the default file path</param>
+        /// <returns></returns>
+        public static K8SConfiguration LoadKubeConfig(string kubeconfigPath = null)
+        {
+            return LoadKubeConfigAsync(kubeconfigPath).GetAwaiter().GetResult();
+        }
+        
+        // <summary>
         ///     Loads Kube Config
         /// </summary>
         /// <param name="kubeconfig">Kube config file contents</param>
         /// <returns>Instance of the <see cref="K8SConfiguration"/> class</returns>
-        private static K8SConfiguration LoadKubeConfig(FileInfo kubeconfig)
+        public static async Task<K8SConfiguration> LoadKubeConfigAsync(FileInfo kubeconfig)
         {
             if (!kubeconfig.Exists)
             {
                 throw new KubeConfigException($"kubeconfig file not found at {kubeconfig.FullName}");
+            }
+
+            using (var stream = kubeconfig.OpenRead())
+            {
+                return await Yaml.LoadFromStreamAsync<K8SConfiguration>(stream);
             }
-
-            var deserializeBuilder = new DeserializerBuilder();
-            var deserializer = deserializeBuilder.Build();
-            using (var kubeConfigTextStream = kubeconfig.OpenText())
-            {
-                return deserializer.Deserialize<K8SConfiguration>(kubeConfigTextStream);
-            }
-        }
-
-        /// <summary>
-        ///     Loads Kube Config from string
-        /// </summary>
-        /// <param name="kubeconfig">Kube config file contents</param>
-        /// <returns>Instance of the <see cref="K8SConfiguration"/> class</returns>
-        private static K8SConfiguration LoadKubeConfig(string kubeconfig)
-        {
-
-            var deserializeBuilder = new DeserializerBuilder();
-            var deserializer = deserializeBuilder.Build();
-            return deserializer.Deserialize<K8SConfiguration>(kubeconfig);
         }
 
         /// <summary>
-        ///     Loads Kube Config from stream.
+        ///     Loads Kube Config
         /// </summary>
         /// <param name="kubeconfig">Kube config file contents</param>
         /// <returns>Instance of the <see cref="K8SConfiguration"/> class</returns>
-        private static K8SConfiguration LoadKubeConfig(Stream kubeconfig)
+        public static K8SConfiguration LoadKubeConfig(FileInfo kubeconfig)

```

Comment:
Let's keep this as a stream, since we want to support loading from network streams as well as files.

Reply:
Good catch. I'll add it back in and use Yaml.LoadFromStreamAsync. Should we keep the method with FileInfo kubeconfig?

Essentially, we would have these method signatures.
```
public static async Task<K8SConfiguration> LoadKubeConfigAsync(string kubeconfigPath = null)
public static K8SConfiguration LoadKubeConfig(string kubeconfigPath = null)
public static async Task<K8SConfiguration> LoadKubeConfigAsync(FileInfo kubeconfig)
public static K8SConfiguration LoadKubeConfig(FileInfo kubeconfig)
public static async Task<K8SConfiguration> LoadKubeConfigAsync(Stream kubeconfigStream)
public static K8SConfiguration LoadKubeConfig(Stream kubeconfigStream)
```

Reply:
Your call about `FileInfo`. it's easy enough to turn a file name into a `Stream`